docs/language/accounts.mdx

```diff
@@ -121,6 +124,11 @@ to the `prepare` phase of the transaction.
       fun getLinkTarget(_ path: CapabilityPath): Path?
       fun unlink(_ path: CapabilityPath)
 
+      // Storage iteration  
+      fun forEachPublic(_ function: ((PublicPath, Type): Bool)): Void
```

Comment:
I understand that only the `Type` is made available here to avoid loading the value. But in the implementation, the value is already loaded anyway (i.e: `storageIterator.Next()` returns the value), so wouldn't it be better to just make the value available here as well? I think loading a value doesn't load the entire value graph (i.e: lazy loading). So it should be fine?

Reply:
The problem with making the value available is that if it is a resource, it needs to be removed and then resaved in order to avoid implicitly duplicating it (which can also re-order things in storage). By just providing the type and then making the developer handle loading resources explicitly we avoid implicit duplication.

Reply:
ah ok, that makes sense!